clang/tools/driver/cc1_main.cpp

```diff
@@ -1358,6 +1359,10 @@ int cc1_main(ArrayRef<const char *> Argv, const char *Argv0, void *MainAddr) {
     }
   }
 
+  // Call this before the Clang pointer is moved below.
+  FinishDiagnosticClient();
+  FinishDiagnosticClientScope.release();
```

Comment:
Nit: would be cleaner to make scope exit instead of call function then release.

Reply:
How would I do that? I prefer to avoid adding new `llvm/ADT` APIs in this branch unless there is no alternative or it would be useful in multiple places.

Reply:
Enclose this part of the code with `{}`? Maybe hold off the indent change so the diff is not huge.

Reply:
This is lose-lose, if I enclose and indent we'll be dealing with merge failures and if don't indent we'll be continuously fighting `clang-format` for this part of the code.

Reply:
I'm fine with the current patch, but if you *did* choose to enclose it and not indent, you could wrap it in marker comments to disable formatting:
```
// clang-format off
...
// clang-format on
```

Reply:
I don't like messing with indentation, doing the call explicitly is not that bad IMO.

Reply:
I am fine as it is now due to conflict potential. Maybe think about `{}` when upstreaming.

Reply:
For upstreaming I would probably move
```
  // When running with -disable-free, don't do any destruction or shutdown.
  if (Clang->getFrontendOpts().DisableFree) {
    llvm::BuryPointer(std::move(Clang));
    return !Success;
  }
```
to its own scope exit ahead of `FinishDiagnosticClientScope`.